### PR TITLE
Adding Dataloader, PersonType, and some data to test it.

### PR DESCRIPTION
**Pull request by DaveyEdwards**
Here is a pull request adding Dataloader to the project.  If you log the response coming out of the getPerson you can see that it is only getting each person once. 

Only thing I am unsure of is on large nested queries it starts to slowdown quite abit, but I think it is the way the 'seed/people' are being referenced in my getPerson function.  Would appreciate any input.  

When I query my db of Google Datastore with this same setup I get amazing performance on 700 lined nested queries.

Here is a gif demonstrating the queries/different sizes with what I have in the pull request
(last query takes around 20seconds to display and about 5min before I can scroll through the data.  It is around 57mb in size)

### CODE REVIEW

**Comment by tim-soft** [comment]
This is great, I was just planning on doing something very similar, so thank you!

What is a little confusing for me here is that you aren't retrieving people by querying Sequelize, which would be the whole point of using DataLoader in practice?

**Comment by DaveyEdwards** [comment]
It never crossed my mind lol.  I haven't used Sequelize in over a year and couldn't see any docs on setting the project up with it.   Unfortunately I will be busy for the next month doing video production work though so I won't be able to get it up and running anytime soon.  If you were wanting to set it up and have questions feel free to ask.

I have a project in heavy development ontop of Google Cloud Platform's Datastore + relay-modern/ graphql connections if your interested in looking at as well (my pull request is pretty much a copy of how dataloader is being used in it).

That repo is here: https://github.com/DaveyEdwards/myiworlds

**Comment by tim-soft** [comment]
Sequelize and an empty sqlite3 database actually come packaged with RSK these days, which is why I brought it up

I posted some code for accessing the db the other day via Sequelize [issues/1296](https://github.com/kriasoft/react-starter-kit/issues/1296) for similar purposes

Querying the database responsibly is something I will need to sort out one way or another. Thanks for the links!

**Comment by DaveyEdwards** [comment]
Implementing this wasn't anywhere near as hard as I had thought, if it helps here is a run down of adding new dataloaders or how you would integrate it with sequelize:

1. Create the new loader inside your loaders folder and put your call to your database inside of it.  Replace the getPerson() function with the function for your database.
```
export default {
  personLoader: new DataLoader(people => Promise.all(people.map(_id => getPerson(_id)) ))
}
```

2. Add a new loader to your loaders/index.js.  This is just to pull in your new loader which your sever.js file grabs to have access to all your loaders.
```
 // Import your new loader
import personLoader from './personLoader';

 // All your dataloaders
 const allLoaders = {
  // Add your new loader here
   ...personLoader
 }
 
 export default function loaders () {
   return allLoaders
 }
```
This is where in the server.js file it is getting those loaders and using them.
```
  pretty: __DEV__,
   context: {
     loaders: loaders()
   },
  })));
```


3. Add dataloader to your schema types.  Your database function is no longer in here but back in your dataloader.
```
   friends: {
      type: new List( PersonType ),
      resolve: async( person, args, { loaders }) => {
        return await loaders.personLoader.loadMany( person.friends );
      }
    }
```
That's it.  Some documentation like this would have been extremely helpful for people like me just last week lol.  If this gets accepted or gets some changes I think I will re write this with better examples to add to the docs, maybe do a video to

File: package.json [hunk 1]
```diff
@@ -19,6 +19,10 @@
     "classnames": "^2.2.5",
     "cookie-parser": "^1.4.3",
     "core-js": "^2.4.1",
+    "dataloader": "^1.3.0",
+    "express": "^4.15.2",
+    "express-graphql": "^0.6.4",
+    "dataloader": "^1.3.0",
```
Comment (embargo2710) [inline package.json]: dataloader two times

**Comment by ulani** [comment]
@DaveyEdwards thank you very much for this PR! Unfortunately, we have close it due to inactivity. Feel free to re-open it or [join](https://discord.com/invite/2nKEnKq) our [Discord](https://discord.com/invite/2nKEnKq) channel for [discussion](https://github.com/kriasoft/react-starter-kit/discussions/1950).

NOTE: The `main` branch has been updated with React Starter Kit v2, using JAM-style architecture.
